elastic/kibana
# [Reporting/Chromium] Update Puppeteer to 10.2.0, use downloaded Chromium for Mac x64/arm

## Summary

This PR updates the Puppeteer dependency for Reporting to 10.2.0, and while doing so, switches the source of Chrome software for Mac to pull the download from https://commondatastorage.googleapis.com/chromium-browser-snapshots/index.html 

Making the switch to use the "off-the-shelf" Chromium software for Mac allows us to easily add support for Mac on ARM architecture since there is no additional custom build for Elastic to create.

Closes https://github.com/elastic/kibana/issues/100429
Closes https://github.com/elastic/kibana/issues/109949
Closes https://github.com/elastic/kibana/issues/109545

### Testing
1. Create a build of Kibana from this branch: `yarn kbn build --skip-os-packages`
2. **Download the build from a browser on a Mac** and unpack the contents 
3. Run these commands to turn off Gatekeeper's extended attributes in Mac OS:
   ```
   xattr -d -r com.apple.quarantine node_modules
   xattr -d -r com.apple.quarantine x-pack/plugins/reporting/chromium
   ```
4. Run `./bin/kibana`
5. Kibana debug logs should show that the Chromium browser binary has been found and registered. From the time to initialize the Reporting plugin to `Start complete`, these should be the only Reporting-specific logs:
    ```
    [DEBUG][plugins.reporting] Initializing plugin
    [DEBUG][plugins.reporting.config] Running on OS: 'Darwin'
    [INFO ][plugins.reporting.config] Chromium sandbox provides an additional layer of protection, and is supported for Darwin OS. Automatically enabling Chromium sandbox.
    [DEBUG][plugins.reporting] Reporting roles configuration is disabled. Please assign access to Reporting use Kibana feature controls for applications.
    [INFO ][plugins.reporting] Browser executable: /Users/sullivat/build/kibana-8.0.0-SNAPSHOT/x-pack/plugins/reporting/chromium/chrome-mac/Chromium.app/Contents/MacOS/Chromium
    [DEBUG][plugins.reporting] Setup complete
    [DEBUG][plugins.reporting.reports:monitor] Task to monitor for pending reports to run every 3s.
    [INFO ][plugins.reporting.store] Creating ILM policy for managing reporting indices: kibana-reporting
    [DEBUG][plugins.reporting] Start complete
    ```
6. Start a Trial license in Stack Management > License Management
7. Create a few PDF and PNG exports in either Canvas, Visualize app, or Dashboard

### TODO
- [x] Create a custom build for Linux x64/arm and update `revision: 856583` in the code
